src/targets/gpu/fuse_ops.cpp

```diff
@@ -785,6 +785,34 @@ struct find_layernorm_pointwise
     }
 };
 
+struct find_concat_pointwise
+{
+    auto matcher() const
+    {
+        return precompile_name("pointwise")(
+            match::arg(0)(precompile_name("concat").bind("concat")));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto ins    = r.result;
+        auto concat = r.instructions["concat"];
+        auto* pm    = ins->module_inputs().front();
+
+        if(not concat->module_inputs().empty())
+            return;
+
```

Comment:
It doesn't look like pm depends on module inputs being empty. 

Swap the creation of PM with the early return here. No point in getting the front() if its not empty.

```
       if(not concat->module_inputs().empty())
            return;
       auto* pm    = ins->module_inputs().front();

```

Reply:
Done